devops/ci.yml

```diff
@@ -94,6 +103,72 @@ jobs:
     workingDirectory: build
     displayName: 'Tests'
 
+- job:
+  displayName: LinuxSandboxTests
+  dependsOn: ConfigureCIRun
+  condition: eq(dependencies.ConfigureCIRun.outputs['setVarStep.testProcSandbox'], 'On')
+  pool:
+    vmImage: 'ubuntu-18.04'
+  timeoutInMinutes: 120
+  strategy:
+    matrix:
+      GCC Release:
+        CC: gcc
+        CXX: g++
+        BuildType: Release
+      GCC Debug:
```

Comment:
We don't need all of these variations for CI, only one of them. Better move the rest to `nightly`.

Reply:
I've removed the GCC versions because GCC is too broken.  The compile + test for the others takes <1 minute each and is run only when the code changes, so I don't see any advantage in moving it to nightly.

Reply:
The main problem is that it starts in a new VM, so it uses more resources. We currently have a limit of 10 concurrent VMs and the CI uses 4, so we can do 2.5 builds in parallel (multiple changes, cancellation of previous builds, etc). If the number of VMs passes 5, we can't even do 2 at a time.

What we left in CI is the minimal configs to catch most errors, so one per type (gcc, clang, win, mac). Everything else goes into the nightly which can take as long as it takes.

Also, the GCC build will have to pass once we move this into `src` if we want to keep GCC support at all. Ignoring it now may make it a lot harder to enable it back later, as we've seen in the runtime last time we enabled it.